patches/server/0977-Add-PlayerPreRespawnLocationEvent.patch

```diff
@@ -0,0 +1,35 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cryptite <cryptite@gmail.com>
+Date: Mon, 1 May 2023 15:23:34 -0500
+Subject: [PATCH] Add PlayerPreRespawnLocationEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
+index 4d837c1530a3031a4c2a5a39d87bd013d60e14a6..203d43c0902132bed1e396bc956d2313447d3534 100644
+--- a/src/main/java/net/minecraft/server/players/PlayerList.java
++++ b/src/main/java/net/minecraft/server/players/PlayerList.java
+@@ -835,6 +835,14 @@ public abstract class PlayerList {
+ 
+         // CraftBukkit start - fire PlayerRespawnEvent
+         if (location == null) {
++            // Paper start
++            Player respawnPlayer = entityplayer1.getBukkitEntity();
++            io.papermc.paper.event.player.PlayerPreRespawnLocationEvent preRespawnEvent = new io.papermc.paper.event.player.PlayerPreRespawnLocationEvent(respawnPlayer, reason);
++            preRespawnEvent.callEvent();
```

Comment:
I think
```patch
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 4d837c153..f9040dafe 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -833,6 +833,14 @@ public abstract class PlayerList {
         boolean isLocAltered = false; // Paper - Fix SPIGOT-5989
         // Paper end
 
+        // Paper start
+        if (location ==  null) {
+            Player respawnPlayer = entityplayer1.getBukkitEntity();
+            io.papermc.paper.event.player.PlayerPreRespawnLocationEvent preRespawnEvent = new io.papermc.paper.event.player.PlayerPreRespawnLocationEvent(respawnPlayer, reason);
+            preRespawnEvent.callEvent();
+            location = preRespawnEvent.getRespawnLocation();
+        }
+        // Paper stop
         // CraftBukkit start - fire PlayerRespawnEvent
         if (location == null) {
             // boolean isBedSpawn = false; // Paper - moved up
```

seems like a potentially cleaner patch. In general, I am unsure if the name of the event really makes sense yet, nor do I see much use in the event beyond the mentioned performance improvement that is gained by not having to compute a safe respawn location.